website/docs/developer/development-guide/code/best-practices.mdx

```diff
@@ -0,0 +1,72 @@
+---
+id: best-practices
+title: Best Practices
+description: Best practices for coding in ORCA.
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+## Unit Testing
+All code should reach minimum 80% coverage through [Unit Tests](unit-tests.md).
+
+## Code Style
+We use the [Google Style Guide](https://google.github.io/styleguide/pyguide.html) for style elements such as documentation, titling, and structure.
+
+## Stop on Failure
+Failures within ORCA break through to the Cumulus workflow they are a part of.
+To this end, raising an error is preferred over catching the error and returning a null value or error message.
+
+```python
+try:
+    value = function(param)
+except requests_db.DatabaseError as err:
+    logging.error(err)
+    raise
+```
+```python
+if not success:
+    logging.error(f"You may log additional information if desired. "
+                  f"param: {param}")
+    raise DescriptiveErrorType(f'Error message to be raised info Cumulus workflow.'
+```
+Retries can then be configured in the workflow json if desired. See
+[documentation](https://docs.aws.amazon.com/step-functions/latest/dg/concepts-error-handling.html)
+and
+[tutorials](https://aws.amazon.com/getting-started/hands-on/handle-serverless-application-errors-step-functions-lambda/)
+for more information.
+```json
+"SomeState": {
```

Comment:
Replace this with a more real wold example like the copy_to_glacier from our test ingest flow.

```json
{
  "CopyToGlacier": {
    "Parameters": {
      "cma": {
        "event.$": "$",
        "task_config": {
          "granules": "{$.meta.processed_granules}",
          "buckets": "{$.meta.buckets}",
          "provider": "{$.meta.provider}",
          "collection": "{$.meta.collection}",
          "files_config": "{$.meta.collection.files}"
        }
      }
    },
    "Type": "Task",
    "Resource": "${copy_to_glacier_task_arn}",
    "Retry": [
      {
        "ErrorEquals": [
          "Lambda.ServiceException",
          "Lambda.AWSLambdaException",
          "Lambda.SdkClientException"
        ],
        "IntervalSeconds": 2,
        "MaxAttempts": 6,
        "BackoffRate": 2
      }
    ],
    "Next": "WorkflowSucceeded"
  },
  "WorkflowFailed": {
    "Type": "Fail",
    "Cause": "Workflow failed"
  },
  "WorkflowSucceeded": {
    "Type": "Succeed"
  }
}
```

Reply:
I stripped out the parameters section. If you feel strongly then I can put it back in, but I don't want it to be difficult to find the portion we're talking about.

Reply:
I am ok with stripping it out. Note my other comment about adding in the "Catch" section that I forgot.